xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/main/java/org/xwiki/mail/script/MailSenderScriptService.java

```diff
@@ -274,6 +275,24 @@ private MailListener getListener(String hint) throws MessagingException
         return listener;
     }
 
+    /**
+     * Obfuscate the email for display purposes, if configured to do so.
+     * @param emailAddress the email address to be obfuscated
+     * @return the obfuscated address
+     * @since 11.1
+     * @since 10.11.4
+     */
+    public String obfuscateForDisplay(final String emailAddress)
+    {
+        String obfuscatedEmailAddress;
+        if (emailAddress != null && this.generalConfiguration.isObfuscateEmails()) {
+            obfuscatedEmailAddress = emailAddress.replaceAll("^(.).*@", "$1...@");
+        } else {
+            obfuscatedEmailAddress = emailAddress;
+        }
+        return obfuscatedEmailAddress;
```

Comment:
We already have an email obfuscator at https://github.com/xwiki/xwiki-platform/blob/e500f31b9b22962519528b7932beef2cf7f23789/xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-general/src/main/java/org/xwiki/mail/EmailAddressObfuscator.java#L33

Reply:
Also it's already provided in https://github.com/xwiki/xwiki-platform/blob/fb8e06afa1a7e46e33ce311eb4b09be81b0567f6/xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-general/src/main/java/org/xwiki/mail/script/GeneralMailScriptService.java#L39